### PR TITLE
basic file upload component

### PR DESCRIPTION
**Pull request by rvanheest**
~~Contribution to #88 with~~ a slightly modified version of the `react-file-loader`.

**Some notes on this:**
* [x] Currently it still uses the deprecated `UNSAFE_componentWillReceiveProps` from React's `Component`. We need to look into this and refactor this method, as this will not be supported anymore in React 17.
    * This was resolved by using some magic in `componentDidUpdate`.
* [x] The 'cancel' button does not work.
    * The original cause of this not working was resolved. We need to further test if the upload is in fact canceled on the server side (also in combination with `easy-deposit-api`)
    * ~~A callback needs to be added for this to trigger an update in the parent `Component`~~
* [x] I do not yet understand fully how (and mainly **why**) the upload functionality works. This is also why I have not yet got the 'cancel' button to work.
    * We understand it now, thanks to Dirk 😃
* ~~The test server only has a `/upload` path added for testing. This one should be replaced with an actual implementation of the `POST /deposit/:id/file/:dir_path`.~~
* Neither the `POST /deposit/:id/file/:dir_path` servlet, nor the `FileLoader` component mimic/adhere to the response given by the actual `easy-deposit-api`.
    * Waiting for the specs to update, out of scope
* [x] Currently the file loader only accepts only certain types of files. Should we keep this whitelist, or make it into a blacklist, or remove this whole barrier?
    * I added an optional whitelist and blacklist option. If it turns out we don't need it, we can always remove it later.
* ~~I just copied/transformed the SCSS from the original project to our own CSS. However, this should match the EASY style guide in the future.~~
    * ~~out of scope~~
* [x] change CSS to match the EASY style guide

**Outside of the scope for this PR:**
* integrate with the file overview
* make mockserver according to the specs of `easy-deposit-api`

~~I'll try to schedule something with Dirk to go over some of these points (mainly 1, 2 and 3).
In the meantime, @lindareijnhoudt please merge this PR into yours, such that you can continue working on it a bit as well. Perhaps you could take some of these remaining points?~~

### CODE REVIEW

**Comment by janvanmansum** [comment]
@rvanheest : if it is work in progress, please add a `- [ ]`  to the cover message.

**Comment by rvanheest** [comment]
@janvanmansum as you can see, this is NOT a PR into master, but rather into a PR from @lindareijnhoudt (#88). Unfortunately this shows up in our overview of all PRs...

**Comment by rvanheest** [comment]
@janvanmansum what I'm trying to say: you're right, this is work in progress, but we need to decide how to proceed from here. This will become clear in the next couple of days, after having talked to Dirk.

**Comment by janvanmansum** [comment]
@rvanheest : no problem, the checkboxes are just used to prevent me from merging stuff that is still under consideration.

File: src/main/typescript/components/form/parts/fileUpload/overview/FileLoader.tsx [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+        }, false)
+
+        request.send(formData)
+
+        return request
+    }
+
+    beforeUnloadFn = (e: BeforeUnloadEvent) => {
+        e.preventDefault()
+        return e.returnValue = "Are you sure you want to leave?"
```
[the remark this answers is not in the document]
Comment (rvanheest) [reply src/main/typescript/components/form/parts/fileUpload/overview/FileLoader.tsx]: yup, if you're leaving the page while a file is uploading, you should probably wait until the upload is finished.

File: src/main/typescript/components/form/parts/fileUpload/overview/FileLoader.tsx [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import * as React from "react"
+import { Component } from "react"
+import "../../../../../../resources/css/fileLoader.css"
+
+enum UploadStatus {
+    DONE = "done",
+    CANCELED = "canceled",
```
[the remark this answers is not in the document]
Comment (rvanheest) [reply src/main/typescript/components/form/parts/fileUpload/overview/FileLoader.tsx]: This is the `UploadStatus` used to determine (amongst others) in a pattern match which CSS class to use. It's not the CSS class itself.
Usually I don't really pay much attention to the difference between the British flavor and American flavour of English. But I'll change this here and in the CSS to `cancelled`.

File: src/main/typescript/components/form/parts/fileUpload/overview/FileLoaderDemo.tsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+            <>
+                <h2 style={{ textAlign: "center" }}>Demo Of File Loader</h2>
+                <input type="file" onChange={this.uploadFile}/>
+                <FileLoader
+                    file={this.state.uploadingFile || null}
+                    url={this.state.uploadingFile ? this.props.fileUploadUrl(this.state.uploadingFile.name) : "#"}
+                    preventReload
+                    showCancelBtn
+                    validFileTypes={{
+                    //     blacklist: ["image/jpeg", "image/png", "video/mp4"],
```
[the remark this answers is not in the document]
Comment (rvanheest) [reply src/main/typescript/components/form/parts/fileUpload/overview/FileLoaderDemo.tsx]: might be. perhaps we want to also block `*.zip` files in the first version?

**Pull request merged**